tests/std/tests/P0896R4_ranges_alg_includes/test.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+using P = pair<int, int>;
+
+struct instantiator {
+    static constexpr P haystack_elements[]     = {{0, 0}, {0, 1}, {1, 0}, {2, 0}, {2, 1}, {4, 0}, {4, 1}};
+    static constexpr int needle_elements[]     = {-1, 0, 3, 3};
+    static constexpr int bad_needle_elements[] = {-1, 0, 2, 2};
+
+    static constexpr auto add_one = [](const int x) { return x + 1; };
+
+    template <ranges::input_range Haystack, ranges::input_range Needle>
+    static constexpr void call() {
+        using ranges::includes, ranges::less;
+
+        { // Validate range overload
+            Haystack haystack{haystack_elements};
+            Needle needle{needle_elements};
+            const same_as<bool> auto result = includes(haystack, needle, ranges::less{}, get_first, add_one);

```

Comment:
You learn new something every day. I assume this is to prevent implicit conversions to bool. 

Neat!

Reply:
Yeah, I would have been using _type-constraint_ s like this instead of `static_assert(same_as<decltype(/*...*/), // ...` all along but I thought MSVC hadn't implemented them yet.

Reply:
> I thought MSVC hadn't implemented them yet.

Apparently this is new for 16.7; I have no idea how they managed to sneak it past me.